cmd/lambda/getclaim/main.go

```diff
@@ -2,20 +2,48 @@ package main
 
 import (
 	"context"
-	"net/http"
 
+	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
 	"github.com/awslabs/aws-lambda-go-api-proxy/httpadapter"
 	"github.com/storacha/indexing-service/pkg/aws"
 	"github.com/storacha/indexing-service/pkg/server"
+	"github.com/storacha/indexing-service/pkg/telemetry"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
 )
 
 func main() {
-	config := aws.FromEnv(context.Background())
-	service, err := aws.Construct(config)
+	cfg := aws.FromEnv(context.Background())
+
+	// an empty API key disables instrumentation
+	if cfg.HoneycombAPIKey != "" {
```

Comment:
noticing this code is essentially shared between every command. you could probably define a withTelemetry command that took a makeHandler and a cfg and then just ran this conditional.

Reply:
good point! I'll refactor it in my next PR.